callbacks/opentelemetry.go

```diff
@@ -0,0 +1,288 @@
+package callbacks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/schema"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type OpenTelemetryCallbacksHandler struct {
+	tracer trace.Tracer
+	opts   OpenTelemetryCallbacksHandlerOptions
+}
+
+// Span attributes should be named per semantic conventions where possible https://opentelemetry.io/docs/specs/semconv/gen-ai/llm-spans/
```

Comment:
```suggestion
// Span attributes should be named per semantic conventions where possible https://opentelemetry.io/docs/specs/semconv/gen-ai/gen-ai-spans/
```

Reply:
Semantic conventions for GenAI are not yet available in the [semconv package](https://pkg.go.dev/go.opentelemetry.io/otel/semconv), but once they are, you should probably rely on that to avoid using invalid names.

Reply:
I don't know if this is typical, but since things can be drifty I was wondering if the link shouldn't be pointing to the version we are using.. @dmathieu wdyt?

```suggestion
// Span attributes should be named per semantic conventions where possible https://github.com/open-telemetry/semantic-conventions/blob/v1.27.0/docs/gen-ai/gen-ai-spans.md
```

Reply:
Yeah, looks like things have already drifted a little on the conventions side since this PR was opened. Seems like a good idea to point to a specific version.

Reply:
> Semantic conventions for GenAI are not yet available in the [semconv package](https://pkg.go.dev/go.opentelemetry.io/otel/semconv), but once they are, you should probably rely on that to avoid using invalid names.

Agree, we should track this with an issue.